mdata/notifier.go

```diff
@@ -84,16 +70,26 @@ func (cl Notifier) Handle(data []byte) {
 					continue
 				}
 			}
-			if cl.CreateMissingMetrics {
+			if createMissing {
+				// support pre-8f800f344f7c9906f32a819134224992931e3a5d producers that don't set the Name property
+				if c.Name == "" {
+					def, ok := idx.Get(key[0])
+					if !ok {
+						log.Error(3, "cluster handler: failed to lookup metricDef with id %s", c.Key)
+						return
+					}
+					c.Name = def.Name
+				}
+
 				schemaId, _ := MatchSchema(c.Name)
 				aggId, _ := MatchAgg(c.Name)
-				agg := cl.Metrics.GetOrCreate(key[0], c.Name, schemaId, aggId)
+				agg := metrics.GetOrCreate(key[0], c.Name, schemaId, aggId)
 				if len(key) == 3 {
 					agg.(*AggMetric).SyncAggregatedChunkSaveState(c.T0, consolidator, uint32(aggSpan))
 				} else {
 					agg.(*AggMetric).SyncChunkSaveState(c.T0)
 				}
-			} else if agg, ok := cl.Metrics.Get(key[0]); ok {
+			} else if agg, ok := metrics.Get(key[0]); ok {
```

Comment:
not really part of this PR, but line `87` - `97` seem like they could be refactored to reduce duplication like (if `agg` is declared at one larger scope):
```
			} else if agg, ok := metrics.Get(key[0]); !ok {
				continue
			}
			if len(key) == 3 {
				agg.(*AggMetric).SyncAggregatedChunkSaveState(c.T0, consolidator, uint32(aggSpan))
			} else {
				agg.(*AggMetric).SyncChunkSaveState(c.T0)
			}
```